research-software-directory/RSD-as-a-service
# 420 drag and drop for React 18 (lib replacement)

# Replacing react-beautiful-dnd with dnd-kit

Closes #420

Changes proposed in this pull request:
* Replace old lib with limited support for React 18 with dnd-kit in edit software / testimonials and in custom markdown pages
* Remove old dependencies   

How to test:
* `docker-compose build frontend` to build new frontend
* `docker-compose up` to start 
* Software testimonials, create new software or edit existing software page. Navigate to testimonials. Change the positions of items. The save button will be activated after change. Save and preview page. The items should have new order as specified during the edit
* Custom markdown page order. Login with RSD admin user. In my case this professor3 (see what you defined in env file).  Under profile mnu click on the administration link. 
![image](https://user-images.githubusercontent.com/9204081/188173518-87224881-458e-4192-9032-136a568095e1.png)
* If there are no custom markdown pages create 3 pages. Then drag the menu items on the left. The page order should be updated. To update the order in the footer of the page reload the page. The order of menu items on admin page and the links in the footer should be the same. Use F12 to open developer console and confirm no errors are shown. Refresh the page multiple times and confirm that drag and drop functionality still works.

## Example custom page order (after page reload)
![image](https://user-images.githubusercontent.com/9204081/188174269-3ad4ea3e-b4d0-44ad-bf66-9818039ed3e3.png)

PR Checklist:
